### ISSUE TITLE
SSL/Reverse Proxy Troubles

### ISSUE THREAD

**Issue by Snnackss** [Hubro/clipface#10]
I am a total noob at configuring nginx and reverse proxies and even using Docker. (only other time I used it was to install smokeping on a Raspberry Pi). 

I was wondering if there was a way to install an SSL certificate using openssl instead of going through a reverse proxy? I am having so much trouble with nginx. I got the proxy pass to work, but it cannot access any of the video files for whatever reason. 

Honestly if anyone could share how they configured their reverse proxy that'd be great. Really want my links to embed into Discord, but this is not possible without https.

**Comment by Hubro** [comment]
> I am a total noob at configuring nginx and reverse proxies and even using Docker. (only other time I used it was to install smokeping on a Raspberry Pi).
> 
> I was wondering if there was a way to install an SSL certificate using openssl instead of going through a reverse proxy? I am having so much trouble with nginx. I got the proxy pass to work, but it cannot access any of the video files for whatever reason.
> 
> Honestly if anyone could share how they configured their reverse proxy that'd be great. Really want my links to embed into Discord, but this is not possible without https.

I'm currently away from my PC, but I'll update the README with an NGINX example later 🙂

**Comment by Hubro** [comment]
> > > I am a total noob at configuring nginx and reverse proxies and even using Docker. (only other time I used it was to install smokeping on a Raspberry Pi).
> > > I was wondering if there was a way to install an SSL certificate using openssl instead of going through a reverse proxy? I am having so much trouble with nginx. I got the proxy pass to work, but it cannot access any of the video files for whatever reason.
> > > Honestly if anyone could share how they configured their reverse proxy that'd be great. Really want my links to embed into Discord, but this is not possible without https.
> > 
> > 
> > I'm currently away from my PC, but I'll update the README with an NGINX example later slightly_smiling_face
> 
> THANK YOU. You are a savior. I SERIOUSLY appreciate your help. Means a lot. I'll take a look when you post it for sure.

No problem, let me know if you have any issues: https://github.com/Hubro/clipface#nginx-reverse-proxy-with-ssl

**Comment by Snnackss** [comment]
> 
> 
> > THANK YOU. You are a savior. I SERIOUSLY appreciate your help. Means a lot. I'll take a look when you post it for sure.
> 
> No problem, let me know if you have any issues: https://github.com/Hubro/clipface#nginx-reverse-proxy-with-ssl

Thank you so much! Its all "secure" now haha. 

As for the guide, there were two things I had to adjust.

1. When you mentioned creating the dhparam.pem file with openssl, the command wouldn't run unless I removed the asterisk. So I had to run the command like this:
`openssl dhparam -out /my/nginx/config/path/dhparam-2048.pem 2048`

2. I had to change the name of the dhparam.pem in the config file from what you had because using the openssl command you shared generated it with the name "dhparam-2048.pem". So, instead of it looking like this: `ssl_dhparam dhparam.pem;`, I simply change it to this: `ssl_dhparam dhparam-2048.pem;` and it worked.
